dbms/src/Columns/ColumnAggregateFunction.cpp

```diff
@@ -265,26 +265,22 @@ void ColumnAggregateFunction::updateHashWithValue(size_t n, SipHash & hash) cons
     hash.update(wbuf.str().c_str(), wbuf.str().size());
 }
 
-/// NOTE: Highly overestimates size of a column if it was produced in AggregatingBlockInputStream (it contains size of other columns)
+/// The returned size is less than real size. The reason is that some parts of
+/// aggregate function data may be allocated on arenas. One arena is used for
+/// several blocks, and it also may be updated concurrently from other threads,
+/// so we can't know the size of these data.
 size_t ColumnAggregateFunction::byteSize() const
 {
     size_t res = data.size() * sizeof(data[0]);
 
-    for (const auto & arena : arenas)
-        res += arena->size();
-
     return res;
 }
 
-
-/// Like byteSize(), highly overestimates size
+/// Like in byteSize(), the size is underestimated.
```

Comment:
over/under?

Reply:
Yes, it's underestimated now, because we don't account for shared arenas (e.g. those used in Aggregator).